arrow-libs/core/arrow-core-retrofit/src/main/kotlin/arrow/retrofit/adapter/either/networkhandling/CallError.kt

```diff
@@ -0,0 +1,23 @@
+package arrow.retrofit.adapter.either.networkhandling
+
+import java.io.IOException
+
+/**
+ * Error hierarchy when calling remote server.
+ */
+public sealed class CallError
+
+/**
+ * Http request returned an error response.
+ */
+public data class HttpError(val code: Int, val body: String) : CallError()
+
+/**
+ * IO error: no network, socket timeout, malformed JSON etc. Check the [cause] for details.
+ */
+public data class IOError(val cause: IOException) : CallError()
```

Comment:
True, this will depend on the deserializing adapter the programmer is using.
* GSON:
```
    Either.Left(IOError(cause=com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1 path $))
```
* Moshi:
```
    Either.Left(IOError(cause=com.squareup.moshi.JsonEncodingException: Use JsonReader.setLenient(true) to accept malformed JSON at path $))
```
* Kotlinx serialization:
```
Either.Left(UnexpectedCallError(cause=kotlinx.serialization.json.internal.JsonDecodingException: Expected start of the object '{', but had 'EOF' instead
    JSON input: not a valid JSON))
```
So it's impossible to write some generic documentation. I can simply remove this statement.